targets/TARGET_WICED/wiced_port/WICED/RTOS/mbed/WWD/wwd_rtos.cpp

```diff
@@ -0,0 +1,304 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2017 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "wwd_rtos.h"
+#include "wwd_constants.h"
+#include "wwd_assert.h"
+#include "wwd_rtos_interface.h"
+#include "wiced_utilities.h"
+#include "mbed.h"
```

Comment:
should not include this , pull in what is needed rather

Reply:
Above comment got hidden by changes, but question still stands:

Is this a problem in cpp files? There's no risk of namespace polution since it's self contained in the cpp file. If anything this would help protect against filename changes.

Reply:
Including only header files that you need will greatly reduce the rebuild time.

On the first build, dependency files are created. Then on the following builds, if there has not been any changes to those headers that you include, your file is not needed to rebuild.

But if everyone includes "mbed.h" this leads every small change to lead to full rebuild of the whole Mbed OS tree.

Reply:
Fair enough, will update